### ISSUE TITLE
Qt client: don't announce state changes for a user if there is no actual state changes aplyed

### ISSUE THREAD

**Issue by amirmahdifard** [BearWare/TeamTalk5#1572]
### Description

this bug is a bit advance i think, but i'll try to explane it as best as possible: when your changing subscription for a user, if that subscription didn't change anything in that time, stil after you toggled the subscription for the user, nvda is once announcing that user's information as it got some change: for example, let me example something for you to understtand: right now, my user is this in teamtalk: amir Laptop 🎤, 👨 (Administrator)  row 11  1  column 1  level 1 : now, i will toggle a subscription of my self, that won't add anything or remove anything from my user in the list, everything will be the same after i do this subscription.: Subscription "Voice" enabled for amir Laptop amir Laptop changed subscription "Voice" to: On . now, after this, my screen reader announced my name again in the focus like this: amir Laptop 🎤, 👨 (Administrator)  row 11  1  column 1  level 1: so y it did that, because if you look to my user status before the subscription and after the subscription, everything is same: my microphone emoji is there in both, my mail emoji, my name, and my administrator status, all is same in both before and after the subscription: so y NVDA announced my name after i toggled the subscription as something changed in there? now, i will example wich times it should announce the change: it should only announce the change if it is difrent after the subscription than before: in this example, see this user: Television, 👨  row 15  1  column 1  level 1 . now, i will intercept voice of this user: and it is like this: Subscription "Intercept voice" enabled for Television and then NVDA announce the user as this: Television 🎤, 👨  row 15  1  column 1  level 1 . so now see the difrents, this user's status before and after the subscription, before the subscription, it didn't have the microphone emoji because his voice wasn't playing for me, but after the subscription, NVDA announced his name as it is changed and the microphone status is added: so please make it that NVDA shouldn't announce anything after the subscription if user's status is same and didn't changed: i hope you understand my point: Qt client vertion: 5.11.

### Application

- [X] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkClassic
- [ ] TeamTalkServer

### Platform

- [X] Windows
- [ ] macOS
- [ ] Android
- [ ] iOS
- [ ] Linux

### Expected behavior

NVDA should only announce the user's name as it actual had a changed and something is removed or added up on a subscription

### Actual behavior

NVDA is announceing the change up on every subscription, even if there is no changes applyed to the user in that subscription

### Steps to reproduce problem

in a server, toggle someone's subscription, try with the one wich won't apply any change to that user: after you do the subscription, you will hear that NVDA is once announcing that user, like you focused on his user again with your arrow key, but actualy you didn't: i don't know what's causing this to happen.

**Comment by bear101** [comment]
@CoBC can this be marked with "Qt accessibility"? It looks to me like we handle it correctly:
https://github.com/BearWare/TeamTalk5/blob/master/Client/qtTeamTalk/mainwindow.cpp#L6810

**Comment by CoBC** [comment]
Yes, I'll report this to Qt.

**Comment by bear101** [comment]
Maybe it's our own fault. We always call  `slotUpdateTreeWidgetItem(item);`
https://github.com/BearWare/TeamTalk5/blob/master/Client/qtTeamTalk/channelstree.cpp#L1322

We should only do this if text, icons or emojis are changed.

**Comment by CoBC** [comment]
Indeed, well seen. Can you change it? :)

**Comment by amirmahdifard** [comment]
@bear101 when @CoBC asked this from you to change it, you said it's  complicated and need a lot of work: but qt client is in state of development, and this is one of the problems that stops me from using qt because classic doesn't have this problem, and i'm a good bug reporter, so if you fix all problems about qt that i've reported, then you even can remove classic completely in the next fue vertion: so meny of my friends, even my girlfriend is saying that qt client is grate since there so meny more tts messages, and a lot of feechers and we love to use qt, but there some accessability  problems that a little stops us from using qt client, and also, the moveing problem that you can't understand how meny people we selected like classic also make us to use classic client for mooving people to other channels: so please fix this: and @CoBC bro, thank you for fixing some problems about qt that i've reported, there not much problems left, so can you at least try to fix mooving problem if possible, and this one too may be because @bear101 told what is the problem: please answer this comment if possible, and again thank you.

**Comment by amirmahdifard** [comment]
@CoBC lol, i nottised that you always answer when bear101 comments,  and says something, but you don't answer others.

**Comment by beqabeqa473** [comment]
Hi.

It is a bit annoying to hear user state change when you are not focused on tree.
Events from treeview should not be processed when another element, for example chat text input is in the focus.

**Comment by amirmahdifard** [comment]
@beqabeqa473 up on subscription that we do, nvda is announcing that user's state change, even if there is no changes happened: after doing a subscription, the user state change should only be announced if there is an actual changes aplyed: for example, if i press controle 4 on a user, and his microphone emoji is gon for me because i muted them: then in that kase, the screen reader should announce his state change, because, befor pressing controle 4, his state had microphone, but after i pressed controle 4, his microphone is gon, so the screen reader announce for this kase makes sence, but announcing the state change up on every subscription even if there is no changes, doesn't makes any sence but just annoying: i hope this bug gets resolved expectedly, because explaning thie bug is a bit complicated for me, so a developer may be can possibly mis understand some how, and end up fixing it in rong way for example: so i hope that it gets fix as expected, and hopefully a beta vertion we get to test this fix.

**Comment by amirmahdifard** [comment]
@bear101 i trust you that you can fix this one expectedly and grately if you want too: as always i have seen your work, i know that your a complete and good developer: thank you.

**Comment by amirmahdifard** [comment]
@bear101 @CoBC please lissen: you don't need to do that hard work to fix this: now i understand how this bug occurs: i think you by mistake, told to Qt client that if user state is changed, don't announce the state change, but if the user status is not changed, then announce the state change up on a subscription: just enough if you correct this: let me tell you with an example: if you change someone's subscription that his user state will not change: for example,  change someones subscription of user messages because user messages subscription doesn't really change state change for the user: if you do this user messages subscription, then your NVDA will announce that persons state as it got some changes: now, do a subscription that will does a state change for the user: for example, disable or enable someones voice subscription who is in your current channel and his microphone is enabled: then in this kase that this user actually had a state change up on this subscription, nvda doesn't announce his state change after the subscription: so fixing this problem is now more simple i think: you coded somehow that in every kace, if up on a subscription there is no state change, announce the state: but if there is a state change up on the subscription, don't announce the state: so just try to code this one upside down, that if there is no state change, don't announce, and if there is a state change, announce it: then this problem will be solved: thank you for reading.

**Comment by bear101** [comment]
@amirmahdifard Can you try the following version and see if it reduces the number of updates: https://bearware.dk/beta/TeamTalk_v5.14.0.5127_Setup.exe

**Comment by amirmahdifard** [comment]
@bear101 hi,
sorry about late answer, i was away: this satterday, i will be able to try tt 5.14, and i will open a new issue if i see a user state change that shouldn't happen.
also, cc
@beqabeqa473 you mentioned an annoing bug, about user state changes, even when you are not in the channels tree, for example you are in the message edit box: i also faced that, and it's annoying a lot: so until satterday, can you also see and confurm if that has also been resolved properly?
thanks both of you.
